src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -957,15 +951,23 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, Compilation
             {
                 if (analyzedArguments.HasErrors)
                 {
-                    // Errors for arguments have already been reported, except for unbound lambdas.
+                    // Errors for arguments have already been reported, except for unbound lambdas and switch expressions.
                     // We report those now.
                     foreach (var argument in analyzedArguments.Arguments)
                     {
-                        var unboundLambda = argument as UnboundLambda;
-                        if (unboundLambda != null)
+                        switch (argument)
                         {
-                            var boundWithErrors = unboundLambda.BindForErrorRecovery();
-                            diagnostics.AddRange(boundWithErrors.Diagnostics);
+                            case UnboundLambda unboundLambda:
+                                var boundWithErrors = unboundLambda.BindForErrorRecovery();
+                                diagnostics.AddRange(boundWithErrors.Diagnostics);
+                                break;
+                            case BoundTupleLiteral _:
+                                // Tuple literals can contain unbound lambdas or switch expressions.
+                                _ = BindToNaturalType(argument, diagnostics);
+                                break;
+                            case BoundUnconvertedSwitchExpression { Type: { } naturalType } switchExpr:
```

Comment:
>{ } [](start = 74, length = 3)

`object`?

Reply:
The former would give the wrong type to the variable, and the latter would incorrectly let through null.

---
In reply to: [301773105](https://github.com/dotnet/roslyn/pull/37052#discussion_r301773105) [](ancestors = 301773105)